node_modules/oae-principals/tests/test-emails.js

```diff
@@ -0,0 +1,603 @@
+                    assert.ok(!err);
+
+                    // Verify the user's email address is verified
+                    var restContext = TestsUtil.createTenantRestContext(tenant.host, 'users-emails-abc123', 'password');
+                    RestAPI.User.getMe(restContext, function(err, user) {
+                        assert.ok(!err);
+                        assert.strictEqual(user.email, 'foo@users.emails.com');
+                        assert.strictEqual(user.emailVerified, true);
+                        assert.ok(!user.emailChange);
+
+                        // Assert there's a mapping for the email address
+                        PrincipalsTestUtil.assertUserEmailMapping('foo@users.emails.com', [user.id], function() {
+
+                            // Update the email address through a CSV import
+                            PrincipalsTestUtil.importUsers(globalAdminRestContext, tenant.alias, getDataFileStream('users-emails-updated.csv'), 'local', true, function(err) {
+                                assert.ok(!err);
+
+                                RestAPI.User.getMe(restContext, function(err, user) {
+                                    assert.ok(!err);
+                                    assert.strictEqual(user.email, 'bar@users.emails.com');
+                                    assert.strictEqual(user.emailVerified, true);
+                                    assert.ok(!user.emailChange);
+
+                                    // Assert there's a mapping for the new email address
+                                    PrincipalsTestUtil.assertUserEmailMapping('bar@users.emails.com', [user.id], function() {
+
+                                        // Assert there's no mapping for the old email address
+                                        PrincipalsTestUtil.assertUserEmailMapping('foo@users.emails.com', [], function() {
+                                            return callback();
+                                        });
+                                    });
+                                });
+                            });
+                        });
+                    });
+                });
+            });
+        });
+
+        /**
+         * Test that verifies that users updating their email address need to verify it
+         */
+        it('verify users updating their email address need to verify it', function(callback) {
+            TestsUtil.generateTestUsers(camAdminRestContext, 1, function(err, users, simong) {
+                assert.ok(!err);
+
+                var oldEmailAddress = simong.user.email;
+
+                // Sanity-check the email address is verified
+                RestAPI.User.getMe(simong.restContext, function(err, me) {
+                    assert.ok(!err);
+                    assert.strictEqual(me.emailVerified, true);
```

Comment:
`assertEmailAddressVerified` ? I think there are a few places we do this process.